Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +142,42 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            return InsertItemToIndex(array, number, index);
+        }
+
+        public static int[] InsertItemToIndex(int[] array, int number, int index)
+        {
+
+            if (array == null || array.Length == 0)
+            {
+                if (array != null && array.Length == 0 && index != 0)
+                {
+                    return array;
+                }
+                int[] newArrayWithOneItem = new int[1] { number };
+                return newArrayWithOneItem;
+            }
+
+            int arrayLength = array.Length;
+            if (index < 0 || index > arrayLength)
+            {
+                return array;
+            }
+            int[] newArray = new int[arrayLength + 1];
+            int newArrayLength = newArray.Length;
+            for (int i = 0, j = 0; i < newArrayLength; i++)
+            {
+                if (i == index)
+                {
+                    newArray[i] = number;
+                    j = 1;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not update the loop counter 'j' within the loop body.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5036134)